dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/TrackedEntityAttributeValidationHook.java

```diff
@@ -97,6 +104,47 @@ public void validateTrackedEntity( ValidationErrorReporter reporter, TrackedEnti
         OrganisationUnit organisationUnit = context.getOrganisationUnit( trackedEntity.getOrgUnit() );
 
         validateAttributes( reporter, trackedEntity, tei, organisationUnit );
+        validateMandatoryAttributes( reporter, trackedEntity );
+    }
+
+    private void validateMandatoryAttributes( ValidationErrorReporter reporter, TrackedEntity trackedEntity )
+    {
+        TrackedEntityType trackedEntityType = reporter.getValidationContext()
+            .getTrackedEntityType( trackedEntity.getTrackedEntityType() );
+
+        if ( trackedEntityType != null )
+        {
+            Set<String> trackedEntityAttributes = Optional.of( trackedEntity )
```

Comment:
This TEI passed to validation cannot be null, so we could simply get the attributes without the optional

Reply:
you're right, thank you